src/sentry/quotas/redis.py

```diff
@@ -5,18 +5,29 @@
 
 from time import time
 
+from sentry import options
 from sentry.exceptions import InvalidConfiguration
 from sentry.quotas.base import NotRateLimited, Quota, RateLimited
-from sentry.utils.redis import get_cluster_from_options, load_script
+from sentry.utils.redis import get_cluster_from_options, load_script, redis_clusters
 
 is_rate_limited = load_script("quotas/is_rate_limited.lua")
 
 
+def get_dynamic_cluster_from_options(setting, config):
+    cluster_name = config.get("cluster", "default")
+    cluster_opts = options.default_manager.get("redis.clusters").get(cluster_name)
+    if cluster_opts is not None and cluster_opts.get("is_redis_cluster"):
+        return True, redis_clusters.get(cluster_name), config
+
+    return (False,) + get_cluster_from_options(setting, config)
+
+
 class BasicRedisQuota(object):
-    __slots__ = ["key", "limit", "window", "reason_code", "enforce"]
+    __slots__ = ["prefix", "subscope", "limit", "window", "reason_code", "enforce"]
 
-    def __init__(self, key, limit=0, window=60, reason_code=None, enforce=True):
-        self.key = key
+    def __init__(self, prefix, subscope=None, limit=0, window=60, reason_code=None, enforce=True):
```

Comment:
nit: The argument ordering here is weird when thinking about passing positional arguments: this seems like `limit`, `window`, and `reason_code` are going to be passed far more frequently than subscope (I'm not even sure why they have default values to begin with, to be honest, they don't add much value.) Since all of the constructors seem to just use keyword arguments and positional ones, this is probably not a big deal, but still just kind of reads weird in the constructor definition.